### PR TITLE
(21/22) Run snippets

### LINKED ISSUE #21: Hookup Run and Save buttons in My Snippets screen
Right now Run and Save don't do anything. Save should call back to the updateSnippet function in the backend and Run will interface with Judge0

### LINKED ISSUE #22: Connect stdout to the Output field in My Snippets screen
The box for output should print the stdout generated by the code snippet/judge0

### PR DESCRIPTION
**Pull request by osbornnick**
closes #22 and part of #21 (run buttons)

(sorry @ghackford I think these issues overlapped, may have stolen some work from you)
state displayed by disabled loading run button, and errors displayed if received. output border shows success or failure
image

### CODE REVIEW

**Comment by osbornnick** [comment]
Oh I also re-formatted the navigation page. Not really on purpose. I format on save with "prettier" and it just kinda happens

File: src/components/navigation/index.js [hunk 1]
```diff
@@ -3,44 +3,57 @@ import "./navigation.css";
...
+                    );
+                })}
+            </ul>
+            <a
+                href="#"
+                className="mt-2 btn btn-lg btn-primary rounded-pill w-100 fw-bold text-white"
+            >
+                Tuit
+            </a>
+            <button className="mt-2 btn btn-lg btn-info rounded-pill w-100 fw-bold text-white">
```
Comment (osbornnick) [inline src/components/navigation/index.js:52]: this is the only thing that changed, everything else was just re-formatted

**Comment by ghackford** [comment]
This may be a bit nitpicky, but is there a way to allow the output to display multiple lines? It probably doesn't matter too much, but it might be nice?
console_output

**Comment by osbornnick** [comment]
> This may be a bit nitpicky, but is there a way to allow the output to display multiple lines? It probably doesn't matter too much, but it might be nice?

Not nitpicky at all! A very good point. Let me see if its a judge0 limitation

**Comment by osbornnick** [comment]
just confirmed judge0 does this correctly. React probably needs to be told to format it respecting \n chars.

**Comment by osbornnick** [comment]
give me a few hours here, I think the change is minimal but I have to get back to my other computer to do it .. I should really upload that .env file somewhere

**Comment by ghackford** [comment]
I'm hooking up the Save button here since we've already done the Run button. Then we can close #21 fully with this merge

**Comment by osbornnick** [comment]
typing quickly into the codemirror box causes a big slowdown, and lots of css errors that seem to come from the codemirror-react package. I'm looking into it. otherwise I think this is ready for merging!

**Review by ghost — APPROVED** [review]

**Comment by NickyL279** [comment]
I can send it to you, email?

From: Nick Osborn ***@***.***>
Date: Monday, April 18, 2022 at 14:49
To: osbornnick/Tuiter ***@***.***>
Cc: zliu ***@***.***>, Review requested ***@***.***>
Subject: Re: [osbornnick/Tuiter] (21/22) Run snippets (PR #27)

give me a few hours here, I think the change is minimal but I have to get back to my other computer to do it .. I should really upload that .env file somewhere

**Pull request merged**